ext/include/opentelemetry/ext/http/common/url_parser.h

```diff
@@ -209,7 +221,7 @@ class UrlDecoder
         hex[1]      = encoded[++pos];
 
         char *endptr;
-        long value = strtol(hex, &endptr, 16);
+        int value = static_cast<int>(std::strtol(hex, &endptr, 16));
```

Comment:
The `google-runtime-int` rule suggests replacing 'long' with 'int64', which seems overkill. Because two hexadecimal characters give a value in the [0; 255] range, downcasting to `int` is safe.

Reply:
Good point on using the `int` type.

How about `std::strtoi()` then ?

Reply:
It looks like there is only std::atoi() but it can only work with base 10 numbers 

https://en.cppreference.com/w/cpp/string/byte

Reply:
Sorry, typo. I meant `std::stoi`

https://en.cppreference.com/w/cpp/string/basic_string/stol

Reply:
`std::stoi()` is a wrapper around `std::stol()`. It performs additional error checking and throws an exception if something is wrong. We would need a try/catch block in this case.

Reply:
I see, thanks for investigating it.

I like the Decode refactoring better anyway, so this is resolved differently since the code no longer uses `std::strtol`